src/components/AnimalCard/index.tsx

```diff
@@ -1,34 +1,67 @@
 import React, { ReactNode, useState } from 'react';
+import { ActivityIndicator } from 'react-native';
 import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
+
+import animalAPI from '../../services/animal/api';
+
+import { Images, Theme } from '../../constants';
+
 import {
-  CardBody, CardBox, CardHeader, CardImage, CardOptions, CardTitle,
+  CardBody, CardBox, CardHeader, CardImage, CardImageLoading, CardOptions, CardTitle,
 } from './styles';
 
-interface IAnimalCardPet {
-  id: string
-}
-
 interface IAnimalCardProps {
     title: string;
-    imageUrlPromise: Promise<string>;
+    imageUrl: string;
     headerOptions?: ReactNode;
     body?: ReactNode;
     headerBackground: string;
-    pet: IAnimalCardPet;
-    onPress?: (pet: IAnimalCardPet) => void;
+    onPress?: () => void;
 }
 
-export const AnimalCard = ({
-  title, body, headerOptions, imageUrlPromise, headerBackground, pet, onPress,
+const AnimalCard = ({
+  title, body, headerOptions, imageUrl, headerBackground, onPress,
 } : IAnimalCardProps) : JSX.Element => {
-  const [petImage, setPetImage] = useState('');
+  const [petImage, setPetImage] = useState<string | null>(imageUrl);
```

Comment:
Kind of a corner-case, but shouldn't this petImage state be initialized as null? If the imagemUrl given is not null but the requested image does not exist on Firebase, this could cause a bug.

Reply:
Solved